Welcome message: allow Jetpack sites to modify

simison opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Follow up to https://github.com/Automattic/wp-calypso/pull/83238

## Proposed Changes

* Allow Jetpack and Atomic sites edit the welcome message
* Move welcome message be the first. Previously the disabled confirm message was first which doesn't make sense.


<img width="1388" alt="Screenshot 2023-10-26 at 17 39 04" src="https://github.com/Automattic/wp-calypso/assets/87168/a6361e92-848f-43ab-b5c9-d2fc2326e432">


## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Go /settings/newsletter
* Test you can edit the welcome message with simple, atomic and jetpack site
* The setting should get stored

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

simison committed:
Welcome message: allow for all sites to modify

simison commented on client/my-sites/site-settings/settings-newsletter/EmailsTextSetting.tsx:
```diff
@@ -94,6 +71,23 @@ export const EmailsTextSetting = ( { value, disabled, updateFields }: EmailsText
 					{ translate( 'The email sent out when someone follows one of your posts.' ) }
 				</FormSettingExplanation>
+
+				<FormLabel htmlFor="confirmation_email_message">
```
I think we should hide this whole section for people who never edited the message.

Just not sure yet how...

TimBroddin replied:
We could return extra fields in the `subscription_options` field from the `/settings` endpoint, changing [this](https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/jetpack/modules/subscriptions.php#L657-L661) to something like this:

```php
	/**
	 * Return merged `subscription_options` option with module default settings. 
     * Include three extra fields that tell whether any of the fields differ from the default setting.
	 */
	public function get_settings() {
		$default_settings = $this->get_default_settings();
		$current_settings = (array) get_option( 'subscription_options' );

		$result = wp_parse_args( $current_settings, $default_settings );

		// Determine if the current settings differ from the default settings 
		foreach ( $default_settings as $key => $value ) {
			$result[ $key . '_differs_from_default' ] = ( isset( $current_settings[ $key ] ) && $current_settings[ $key ] !== $value );
		}

		return $result;
	}

```

It feels a bit hackish, but it might work 😀

simison replied:
Yeah, not sure if worth effort since we're moving to template based (in block editor) anyway at some point soon.

simison replied:
Thinking if a simple string comparison in the client side would work...

TimBroddin replied:
I think we lack the translations there. We could manually bring them in ourselves, but like you say, it's a lot of effort for something temporary.

TimBroddin commented:
I doesn't save yet on my WoA site. I think Jetpack still needs to be updates there?

simison commented:
Oh, looks like they had to revert weekly update p9o2xV-3yE-p2

For Jetpack we probably need to add version check?

simison commented:
I'll add the check back in this PR so that we can just merge the order change.

simison committed:
Add back simple site check and add JP version check

simison commented:
- Added back simple site check
- Added Jetpack version check for the next release (Nov 6). You can test earlier version by manually adjusting the number.

Do Newsletter categories also need the version check?

TimBroddin commented:
> Do Newsletter categories also need the version check?

No, they'll automatically switch over from using the pills inside the subscribe block, to the pills inside the modal. The modal checks if a query parameter is passed from the block (which happens when there are newsletter categories), and skips the newsletter category step inside the modal accordingly.

TimBroddin reviewed (approved):
Nice! Works as intended.

TIL about `isJetpackMinimumVersion`
